### PR TITLE
feat: Save target platform as part of TRTEngine Metadata

### LINKED ISSUE #3101: ✨[Feature] An a checker in the runtime: get_extra_state/set_extra_state
**Is your feature request related to a problem? Please describe.**
add a target_platform in the serialized_engine_info
for TorchTensorRTModule and PythonTorchTensorRTModule

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #3103: Fix typo in runtime.h
https://github.com/pytorch/TensorRT/blob/6a3864821d385b095db25ec6f4e18128eec493d9/core/runtime/runtime.h#L50

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Saves the target platform as part of the TRTEngine Data. 

Fixes #3103
Fixes #3101

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: core/runtime/Platform.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+Platform get_current_platform() {
+#if defined(__linux__) || defined(__gnu_linux__)
+#if defined(__aarch64__)
+  return Platform(Platform::PlatformEnum::kLINUX_AARCH64);
+#elif defined(__amd64__) || defined(__x86_64__)
+  return Platform(Platform::PlatformEnum::kLINUX_X86_64);
+#else
+  return Platform(Platform::PlatformEnum::kLINUX_X86_64);
```
Comment (narendasan) [inline core/runtime/Platform.cpp:82]: This should be unknown

File: core/runtime/TRTEngine.cpp [hunk 2]
```diff
@@ -196,7 +209,6 @@ TRTEngine::TRTEngine(
 }
 
 TRTEngine::~TRTEngine() {
-  cudagraph.reset();
```
Comment (peri044) [inline core/runtime/TRTEngine.cpp:199]: Why is this not needed ?

**Review by peri044 — APPROVED** [review]
LGTM with pending comments.

File: core/runtime/TRTEngine.cpp (hunk 2 above)
Comment (narendasan) [reply core/runtime/TRTEngine.cpp:199]: The object destructor calls it and its potentially unsafe to do it ourselves

File: py/torch_tensorrt/_enums.py [hunk 3]
```diff
@@ -1328,3 +1328,72 @@ def __eq__(self, other: Union[trt.EngineCapability, EngineCapability]) -> bool:
...
+        if platform.system().lower().startswith("linux"):
+            # linux
+            if platform.machine().startswith("aarch64"):
+                return Platform.LINUX_AARCH64
+            elif platform.machine().startswith("x86_64"):
+                return Platform.LINUX_X86_64
+
+        elif platform.system().lower().startswith("win32"):
+            # Windows...
+            if platform.machine().startswith("x86_64"):
```
Comment (narendasan) [inline py/torch_tensorrt/_enums.py:1380]: .lower arm64

**Pull request merged**